src/server/tran_server.cpp

```diff
@@ -552,33 +552,32 @@ tran_server::connection_handler::disconnect_async (bool with_disc_msg)
   m_disconn_future = std::async (std::launch::async, [this, with_disc_msg]
   {
     on_disconnecting (); // server-type specific jobs before disconnecting.
-    // m_conn is not nullptr since it's only set to nullptr here below once
 
-    m_conn->stop_response_broker (); // wake up threads waiting for a response and tell them it won't be served.
+    /*
+     * Stop incoming communication and wake up threads waiting for a response and tell them it won't be served.
+     * This has to be out of the lock of m_conn_mtx because a waitor may hold the lock.
```

Comment:
Where is the case?

Reply:
in `tran_server::connection_handler::send_receive()`. I'll add it in comment.